### PR TITLE
Rust: Extracted Files diagnostic query

### PR DESCRIPTION
**Pull request by geoffw0**
Adds a `rust/diagnostics/successfully-extracted-files` query and some test cases for this and other diagnostic queries to run against.  I'm not sure why only one of the files is successfully extracted, there ought to be three successes and two failures.  We may need an equivalent of:
```cpp
(cpp)
// semmle-extractor-options: --expect_errors
```
or:
```swift
(swift)
//codeql-extractor-expected-status: 1
```
on the two source files that aren't expected to build successfully?

### CODE REVIEW

File: rust/ql/src/queries/diagnostics/ExtractedFiles.ql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * @kind diagnostic
+ * @id rust/diagnostics/successfully-extracted-files
+ * @tags successfully-extracted-files
+ */
+
+import rust
+
+from File f
+where exists(f.getRelativePath())
+select f, "File successfully extracted."
```
[the remark this answers is not in the document]
Comment (geoffw0) [reply rust/ql/src/queries/diagnostics/ExtractedFiles.ql:13]: It's the same message as CPP and Swift.  The other languages leave the message blank, which I guess is OK but doesn't seem very friendly.

**Comment by geoffw0** [comment]
@redsun82 can you tell what's going on here?  Is it likely to be affected by https://github.com/github/codeql/pull/17445 ?

**Comment by geoffw0** [comment]
... just merged in main after https://github.com/github/codeql/pull/17445 was merged.
- it compiles all five files now.
- plus a generated `lib.rs` apparently; it's weird this is visible, but probably not much of a problem if it only effects tests.
  - weirder still the file is *not* deleted, it hangs around after the test is finished.
- the DBCheck failure still happens.

**Comment by geoffw0** [comment]
I've made a [small change](https://github.com/github/codeql/pull/17442/commits/587ebbffbac5099e81e4a4c792db41815ec5fba8) to the test so that it now avoids triggering the DBCheck issue.  That means this PR should be mergeable now.  If someone would like to investigate the DBCheck failure more closely, it's pretty clear from that change how to reproduce it, but the issue is orthogonal to the new diagnostic query this PR is about.

**Review by aibaars — APPROVED** [review]
Looks good to me!

**Pull request merged**